### PR TITLE
fixes publish button warning for development environment #1928

### LINKED ISSUE #1928: Publish button warns in development mode
Based on the condition at the following line, it seems expected that a prompt should not be shown in development environments:

https://github.com/WordPress/gutenberg/blob/6a8e248/editor/header/tools/publish-button.js#L54

The issue is that the boolean coercion occurs before the equality check, so it effectively becomes `false === 'production'`, which will never short-circuit.

Instead, we should use a `!==` operator.

### PR DESCRIPTION
**Pull request by stephenc222**
Fixes #1928. The window confirmation warning dialog is no longer called when in the development environment.

### CODE REVIEW

File: editor/header/tools/publish-button.js [hunk 1]
```diff
@@ -51,7 +51,7 @@ function PublishButton( {
 	const className = classnames( 'editor-tools__publish-button', { 'is-saving': isSaving } );
 	const onClick = () => {
 		const doSave = isPublished ||
-			! process.env.NODE_ENV === 'production' ||
+			! process.env.NODE_ENV !== 'production' ||
```
Comment (westonruter) [inline editor/header/tools/publish-button.js]: Should this not just be `process.env.NODE_ENV !== 'production'`? There are double negatives which make this confusing. The save should happen automatically (without `confirm`) if not in the production environment.

**Comment by stephenc222** [comment]
Rebased and force pushed with the change to keep this PR to a single commit.

**Comment by aduth** [comment]
Heya @stephenc222, what was the reason for closing this pull request? The changes look good to me.

**Comment by stephenc222** [comment]
@aduth I wasn't sure if it was going to be reviewed again after the changes I made (that I should have caught in the first place) and because it's a pretty easy and simple change, I figure I'd let someone else try it, so they can begin contributing here. That's the only reason, thanks for the feedback!

**Pull request closed without merging**